### PR TITLE
test: add unit tests for CommandExecutor

### PR DESCRIPTION
**Pull request by ThangTM26**
Add unit tests for CommandExecutor to verify correct command dispatching and fallback behavior.

### CODE REVIEW

File: src/test/java/org/example/VChatTalk/command/service/CommandExecutorTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        verify(joinCommand, never()).execute(any(), any());
+    }
+
+    @Test
+    @DisplayName("Returns correct command for given type")
+    void getCommand_ExistingType() {
+        IChatCommand command = commandExecutor.getCommand(CommandType.JOIN);
+
+        assertEquals(joinCommand, command);
+    }
```
[the remark this answers is not in the document]
Comment (ThangTM26) [reply src/test/java/org/example/VChatTalk/command/service/CommandExecutorTest.java:81]: Screenshot 2026-01-07 104742
All missing test cases have been added. 
The new tests cover additional edge cases and all test cases pass successfully.

**Comment by Toffu-BlankieChoccy** [comment]
All good here. Commencing merge :D
